utils/httpstream.go

```diff
@@ -44,17 +47,39 @@ func StartHTTPServer(handler Handler, path, format string) {
 	}
 
 	// try to remove pre-existing UDS: ignore errors here
-	if tokens[0] == "unix" {
-		os.Remove(tokens[1])
+	phonySock := filepath.Join(filepath.Dir(uri.Path), "phony"+filepath.Base(uri.Path))
+	if uri.Scheme == "unix" {
+		os.Remove(phonySock)
+		os.Remove(uri.Path)
 	}
 
-	listener, err := net.Listen(tokens[0], tokens[1])
+	listener, err := net.Listen(uri.Scheme, phonySock)
```

Comment:
i gave up on dancing around what this should be if it's unix or tcp... if it's tcp we don't need the phony sock dance at all, but it won't be a path, anyway I figure we can wait until that bridge is on the horizon...